src/base/net/downloadmanager.h

```diff
@@ -42,16 +46,54 @@ namespace Net
 {
     class DownloadHandler;
 
+    struct DownloadRequest
+    {
+        QString url;
+        bool saveToFile;
+        qint64 limit;
+        bool handleRedirectToMagnet;
+        QString userAgent;
+    };
+
+    class CreateDownloadRequest
```

Comment:
Could you explain, please, why do you put these function into their own class?

Reply:
Just one of possible implementations... So as not to puzzle over the resolution of name conflicts. We can't have both `limit` and `limit()`, for example, right? At least in a convenient way...

Reply:
Um... for some reason I had not considered this option... I thought to keep it as simple struct as it was initially.
Well, I'll remake it.